README.md
If you're looking to **build a website or a cross-platform mobile app** – we will be happy to help you! Send a note to raushanranjan157@gmail.com and we will be in touch with you shortly.

![Sample Package]()

[📲See example app](https://github.com/raushanji/my-sample-package)

# My Sample Package Documentation

## Import components
1. `yarn add @raushanranjan/my-sample-package` OR 
2. `npm i @raushanranjan/my-sample-package`
2. Use with ES6 syntax to import components

```js
import {
  Message,
  Welcome,
} from 'my-sample-package'

```

## Quick Example
```jsx
<View>
  <Text>
    Welcome Screen
  </Text>
  <Welcome
    width={SCREEN_WIDTH}
    height={SCREEN_HEIGHT}
    fontSize={24}
    color={'#000'}
    fontWeight={'bold'}
    textAllign={'center'}
  />
</View>
```

## Welcome Properties Detail
Define a `welcome` screen with following properties as such:


| Property        | Type           | Description  |
| ------------- |-------------| -----|
| width | Number | Defines the width of the view in which welcome screen text will be shown  |
| height | Number | Defines the height of the view in which welcome screen text will be shown  |
| fontSize | Number | Defines the fontSize of the text shown in welcome screen |
| color | string | Defines the color of the text shown in welcome screen  |
| fontWeight | string | Defines the fontWeight`('bold','normal',etc.)` of the text shown in welcome screen |
| textAllign | string | Defines the alignment of the text shown in welcome screen |

## Responsive Welcome Screen 
To render a responsive screen, use `Dimensions` react-native library to get the width of the screen of your device like such
```js
import { Dimensions } from 'react-native'
const screenWidth = Dimensions.get('window').width
```

## Message Screen
```jsx
  <Welcome
    width={SCREEN_WIDTH}
    height={SCREEN_HEIGHT}
    message={'Welcome To My Sample Package'}
    fontSize={24}
    color={'#000'}
    fontWeight={'bold'}
    textAllign={'center'}
  />
```

## Message Screen Properties Detail
Define a `message` screen with following properties as such:


| Property        | Type           | Description  |
| ------------- |-------------| -----|
| width | Number | Defines the width of the view in which welcome screen text will be shown  |
| height | Number | Defines the height of the view in which welcome screen text will be shown  |
| message | string | Defines the message which you want to show |
| fontSize | Number | Defines the fontSize of the text shown in welcome screen |
| color | string | Defines the color of the text shown in welcome screen  |
| fontWeight | string | Defines the fontWeight`('bold','normal',etc.)` of the text shown in welcome screen |
| textAllign | string | Defines the alignment of the text shown in welcome screen |


## Welcome Screen
```jsx
  <Welcome
    width={SCREEN_WIDTH}
    height={SCREEN_HEIGHT}
    fontSize={24}
    color={'#000'}
    fontWeight={'bold'}
    textAllign={'center'}
  />
```

## Welcome Properties Detail
Define a `welcome` screen with following properties as such:


| Property        | Type           | Description  |
| ------------- |-------------| -----|
| width | Number | Defines the width of the view in which welcome screen text will be shown  |
| height | Number | Defines the height of the view in which welcome screen text will be shown  |
| fontSize | Number | Defines the fontSize of the text shown in welcome screen |
| color | string | Defines the color of the text shown in welcome screen  |
| fontWeight | string | Defines the fontWeight`('bold','normal',etc.)` of the text shown in welcome screen |
| textAllign | string | Defines the alignment of the text shown in welcome screen |


CHANGELOG.md
# Changelog

## v1.0.0
- first launching of my-sample-package

App.js
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Fragment} from 'react';
import 'babel-polyfill'
import React from 'react'
import { Dimensions, StyleSheet, ScrollView, View, Text, StatusBar } from 'react-native';
import Message from './src/Message';
import Welcome from './src/Welcome';

const SCREEN_HEIGHT = Dimensions.get('window').height;
const SCREEN_WIDTH = Dimensions.get('window').width;


export default class App extends React.Component {

  render(){
    return (
       <ScrollView>
         <Text style={{textAlign:' center', fontSize: 24, fontWeight: 'bold'}}>Welcome Screen</Text>
         <Welcome
           ViewWidth={SCREEN_WIDTH/1.5}
           ViewHeight={SCREEN_HEIGHT/3}
           fontSize={18}
           TextColor={'pink'}
           FontWeight={'bold'}
           TextAllign={'center'}
         />
         <Text></Text>
         <Text></Text>
         <Text style={{textAlign:' center', fontSize: 24, fontWeight: 'bold'}}>Message Screen</Text>
         <Message
           ViewWidth={SCREEN_WIDTH/1.5}
           ViewHeight={SCREEN_HEIGHT/3}
           Message={'Hello Everybody, Welcome to Sample Package'}
           fontSize={18}
           TextColor={'orange'}
           FontWeight={'bold'}
           TextAllign={'center'}
         />
         <View style={{height: SCREEN_HEIGHT/6}}></View>
       </ScrollView>
    )
  }


}


src/Welcome.js
import React, {Component} from 'react';
import {View, Text, Dimensions} from 'react-native';


const SCREEN_HEIGHT = Dimensions.get('window').height;
const SCREEN_WIDTH = Dimensions.get('window').width;

export default class Welcome extends Component {
    constructor(props){
        super(props);
        this.state = {
            width: this.props.ViewWidth ? this.props.ViewWidth : SCREEN_WIDTH,
            height: this.props.ViewHeight ? this.props.ViewHeight : SCREEN_HEIGHT,
            fontSize: this.props.fontSize ? this.props.fontSize : 24,
            color: this.props.TextColor ? this.props.TextColor : '#000',
            fontWeight: this.props.FontWeight ? this.props.FontWeight : 'bold',
            textAllign: this.props.TextAllign ? this.props.TextAllign : 'center'
        }
    }

    render() {
        return (
            <View style={{width: this.state.width, height: this.state.height, justifyContent: 'center', alignItems: 'center'}}>
                <Text style={{fontSize: this.state.fontSize, fontWeight: this.state.fontWeight, color: this.state.color, textAlign: this.state.textAllign}}>WELCOME TO MY SAMPLE PACKAGE</Text>
                <Text></Text>
                <Text></Text>
                <Text></Text>
                <Text style={{fontSize: this.state.fontSize, fontWeight: this.state.fontWeight, color: this.state.color, textAlign: this.state.textAllign}}>THANKS YOU</Text>
                <Text style={{fontSize: this.state.fontSize, fontWeight: this.state.fontWeight, color: this.state.color, textAlign: this.state.textAllign}}>RAUSHAN RANJAN</Text>
            </View>
        )
    }
}

src/Message.js
import React, {Component} from 'react';
import {View, Text, Dimensions} from 'react-native';


const SCREEN_HEIGHT = Dimensions.get('window').height;
const SCREEN_WIDTH = Dimensions.get('window').width;

export default class Message extends Component {
    constructor(props){
        super(props);
        this.state = {
            width: this.props.ViewWidth ? this.props.ViewWidth : SCREEN_WIDTH,
            height: this.props.ViewHeight ? this.props.ViewHeight : SCREEN_HEIGHT,
            fontSize: this.props.fontSize ? this.props.fontSize : 24,
            color: this.props.TextColor ? this.props.TextColor : '#000',
            fontWeight: this.props.FontWeight ? this.props.FontWeight : 'bold',
            textAllign: this.props.TextAllign ? this.props.TextAllign : 'center',
            message: this.props.Message ? this.props.Message : 'Welcome To Sample Package'
        }
    }

    render() {
        return (
            <View style={{width: this.state.width, height: this.state.height, justifyContent: 'center', alignItems: 'center'}}>
                <Text style={{fontSize: this.state.fontSize, fontWeight: this.state.fontWeight, color: this.state.color, textAlign: this.state.textAllign}}>{this.state.message}</Text>
                <Text></Text>
                <Text></Text>
                <Text></Text>
                <Text style={{fontSize: this.state.fontSize, fontWeight: this.state.fontWeight, color: this.state.color, textAlign: this.state.textAllign}}>Thank you</Text>
                <Text style={{fontSize: this.state.fontSize, fontWeight: this.state.fontWeight, color: this.state.color, textAlign: this.state.textAllign}}>Raushan Ranjan</Text>
            </View>
        )
    }
}